instrumentation/lettuce/lettuce-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v4_0/LettuceSyncClientTest.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.lettuce.v4_0;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchException;
+
+import com.google.common.collect.ImmutableMap;
+import com.lambdaworks.redis.ClientOptions;
+import com.lambdaworks.redis.RedisClient;
+import com.lambdaworks.redis.RedisConnectionException;
+import com.lambdaworks.redis.api.StatefulRedisConnection;
+import com.lambdaworks.redis.api.sync.RedisCommands;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import io.opentelemetry.instrumentation.testing.internal.AutoCleanupExtension;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.SemanticAttributes;
+import java.util.Map;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+@SuppressWarnings("deprecation") // until old http semconv are dropped in 2.0
+class LettuceSyncClientTest {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @RegisterExtension static final AutoCleanupExtension cleanup = AutoCleanupExtension.create();
+
+  static final DockerImageName containerImage = DockerImageName.parse("redis:6.2.3-alpine");
+
+  private static final int DB_INDEX = 0;
+
+  // Disable auto reconnect, so we do not get stray traces popping up on server shutdown
+  private static final ClientOptions CLIENT_OPTIONS =
+      new ClientOptions.Builder().autoReconnect(false).build();
+
+  private static final GenericContainer<?> redisServer =
+      new GenericContainer<>(containerImage).withExposedPorts(6379);
+
+  private static String host;
+  private static int port;
+  private static int incorrectPort;
+  private static String dbUriNonExistent;
+  private static String embeddedDbUri;
+
+  private static final ImmutableMap<String, String> testHashMap =
+      ImmutableMap.of(
+          "firstname", "John",
+          "lastname", "Doe",
+          "age", "53");
+
+  static RedisClient redisClient;
+
+  private static StatefulRedisConnection<String, String> connection;
+  static RedisCommands<String, String> syncCommands;
+
+  @BeforeAll
+  public static void setUp() {
```

Comment:
Nit: JUnit 5 test lifecycle methods need not be public
```suggestion
  static void setUp() {
```